Using `=` inside f-strings breaks pyflyby's parsing

Given the following code block, I would expect pyflyby to `import math`, but it doesn't import anything.

```
x = 1
f"{x=}"
math.sqrt(x)
```

The only solution I have is to remove the `=` and use `f"x={x}" instead.

This is the error I get on Linux in Python 3.13.9 (the feature was introduced in 3.8), using pyflyby v1.10.1 (installed with uv):
```
Traceback (most recent call last):
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_cmdline.py", line 394, in process_actions
    action(m)
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_cmdline.py", line 436, in action_ifchanged
    if m.output_content.joined == m.input_content.joined:
       ^^^^^^^^^^^^^^^^
  File "/home/wip/anaconda3/lib/python3.12/functools.py", line 993, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_cmdline.py", line 349, in output_content
    return FileText(self.modifier(self.input_content), filename=self.filename)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/bin/tidy-imports", line 132, in modify
    block = fix_unused_and_missing_imports(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_imports2s.py", line 374, in fix_unused_and_missing_imports
    _codeblock = reformat_import_statements(_codeblock, params=params)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_imports2s.py", line 296, in reformat_import_statements
    transformer = SourceToSourceFileImportsTransformation(codeblock)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_imports2s.py", line 29, in __new__
    return cls._from_source_code(arg)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_imports2s.py", line 45, in _from_source_code
    self.preprocess()
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_imports2s.py", line 114, in preprocess
    for is_imports, subblock in self.input.groupby(lambda ps: ps.is_import):
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_parse.py", line 1224, in groupby
    for pred, stmts in groupby(self.statements, predicate):
                               ^^^^^^^^^^^^^^^
  File "/home/wip/anaconda3/lib/python3.12/functools.py", line 993, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_parse.py", line 1144, in statements
    node = self.annotated_ast_node
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/anaconda3/lib/python3.12/functools.py", line 993, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_parse.py", line 1065, in annotated_ast_node
    return _annotate_ast_nodes(result)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_parse.py", line 348, in _annotate_ast_nodes
    _annotate_ast_startpos(aast_node, None, startpos, text, flags)
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_parse.py", line 414, in _annotate_ast_startpos
    leftstr = _annotate_ast_startpos(
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_parse.py", line 414, in _annotate_ast_startpos
    leftstr = _annotate_ast_startpos(
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/wip/.local/share/uv/tools/pyflyby/lib/python3.12/site-packages/pyflyby/_parse.py", line 421, in _annotate_ast_startpos
    raise AssertionError(
AssertionError: While processing /home/wip/projects/ai_bazaar/sandboxes/chat_service.py: Got out-of-order AST node(s):
  parent minpos=(1,1)
    node: JoinedStr(values=[Constant(value='x='), FormattedValue(value=Name(id='x', ctx=Load()), conversion=114)])
      fields: values
      children:
                (2,4): Constant(value='x=')
        ==>     (2,3): FormattedValue(value=Name(id='x', ctx=Load()), conversion=114)

This indicates a bug in pyflyby._

pyflyby developer: Check if there's a bug or missing ast node handler in pyflyby._parse._iter_child_nodes_in_order() - probably the handler for ast.JoinedStr.

/home/wip/.local/share/uv/tools/pyflyby/bin/tidy-imports: encountered the following problems:
    /home/wip/projects/ai_bazaar/sandboxes/chat_service.py: AssertionError: Got out-of-order AST node(s):              parent minpos=(1,1)
                node: JoinedStr(values=[Constant(value='x='), FormattedValue(value=Name(id='x', ctx=Load()), conversion=114)])
                  fields: values
                  children:
                            (2,4): Constant(value='x=')
                    ==>     (2,3): FormattedValue(value=Name(id='x', ctx=Load()), conversion=114)
            
            This indicates a bug in pyflyby._
            
            pyflyby developer: Check if there's a bug or missing ast node handler in pyflyby._parse._iter_child_nodes_in_order() - probably the handler for ast.JoinedStr.
```

--- a/lib/python/pyflyby/_parse.py
+++ b/lib/python/pyflyby/_parse.py
@@ -123,6 +123,8 @@ def _iter_child_nodes_in_order(node):
 
     ``ast.iter_child_nodes`` does the same thing, but not in source order.
     e.g. for ``Dict`` s, it yields all key nodes before all value nodes.
+
+    `JoinedStr` for `f"{x=}"` also does not.
     """
     return _flatten_ast_nodes(_iter_child_nodes_in_order_internal_1(node))
 
@@ -207,6 +209,12 @@ def _iter_child_nodes_in_order_internal_1(node):
     elif isinstance(node, ast.FormattedValue):
         assert node._fields == ('value', 'conversion', 'format_spec')
         yield node.value,
+    elif isinstance(node, ast.JoinedStr):
+        assert node._fields == ("values",)
+        # Sort values by their position in the source code
+        # for f"{x=}", nodes are not in order
+        sorted_children = sorted(node.values, key=lambda v: (v.lineno, v.col_offset))
+        yield sorted_children
     elif isinstance(node, MatchAs):
         yield node.pattern
         yield node.name,

--- a/lib/python/pyflyby/_parse.py
+++ b/lib/python/pyflyby/_parse.py
@@ -42,6 +42,11 @@ else:
     from types import NoneType
     from ast import MatchAs, MatchMapping
 
+if sys.version_info >= (3, 14):
+    from ast import TemplateStr
+else:
+    TemplateStr = None  # type: ignore
+
 
 def _is_comment_or_blank(line, /):
     """
@@ -209,10 +214,12 @@ def _iter_child_nodes_in_order_internal_1(node):
     elif isinstance(node, ast.FormattedValue):
         assert node._fields == ('value', 'conversion', 'format_spec')
         yield node.value,
-    elif isinstance(node, ast.JoinedStr):
+    elif isinstance(node, ast.JoinedStr) or (
+        TemplateStr is not None and isinstance(node, TemplateStr)
+    ):
         assert node._fields == ("values",)
         # Sort values by their position in the source code
-        # for f"{x=}", nodes are not in order
+        # for f"{x=}" / t"{x=}", nodes are not in order
         sorted_children = sorted(node.values, key=lambda v: (v.lineno, v.col_offset))
         yield sorted_children
     elif isinstance(node, MatchAs):

--- a/tests/test_parse.py
+++ b/tests/test_parse.py
@@ -1498,6 +1498,31 @@ def test_join_formatted_string_columns(input):
 @pytest.mark.parametrize(
     "input",
     [
+        '''
+x = 42
+y = f'{x=}'
+''',
+        '''
+a = 1
+b = 2
+result = f'{a=} {b=}'
+''',
+        '''
+name = 'Alice'
+age = 30
+info = f'{name=}, {age=}'
+''',
+    ],
+)
+def test_fstring_debug_expressions(input):
+    """Test that f-string debug expressions (f'{x=}') are handled correctly."""
+    block = PythonBlock(input, auto_flags=True)
+    assert block.annotated_ast_node
+
+
+@pytest.mark.parametrize(
+    "input",
+    [
         '''b"""
 two
 """ b"""

--- a/tests/test_parse.py
+++ b/tests/test_parse.py
@@ -1525,6 +1525,32 @@ def test_fstring_debug_expressions(input):
     assert block.annotated_ast_node
 
 
+@pytest.mark.skipif(sys.version_info < (3, 14), reason="Template strings require Python 3.14+")
+@pytest.mark.parametrize(
+    "input",
+    [
+        """
+x = 42
+y = t"{x=}"
+""",
+        """
+a = 1
+b = 2
+result = t"{a=} {b=}"
+""",
+        """
+name = 'Alice'
+age = 30
+info = t"{name=}, {age=}"
+""",
+    ],
+)
+def test_template_string_debug_expressions(input):
+    """Test that template string debug expressions (t'{x=}') are handled correctly."""
+    block = PythonBlock(input, auto_flags=True)
+    assert block.annotated_ast_node
+
+
 @pytest.mark.parametrize(
     "input",
     [

--- a/tests/test_parse.py
+++ b/tests/test_parse.py
@@ -1498,20 +1498,25 @@ def test_join_formatted_string_columns(input):
 @pytest.mark.parametrize(
     "input",
     [
-        '''
+        """
 x = 42
 y = f'{x=}'
-''',
-        '''
+""",
+        """
 a = 1
 b = 2
 result = f'{a=} {b=}'
-''',
-        '''
+""",
+        """
+a = 1
+b = 2
+result = f'{a+b=}'
+""",
+        """
 name = 'Alice'
 age = 30
 info = f'{name=}, {age=}'
-''',
+""",
     ],
 )
 def test_fstring_debug_expressions(input):

